src/main/java/htsjdk/samtools/SAMTextHeaderCodec.java

```diff
@@ -124,23 +123,14 @@ public SAMFileHeader decode(final LineReader reader, final String source) {
         mFileHeader.setSequenceDictionary(new SAMSequenceDictionary(sequences));
         mFileHeader.setReadGroups(readGroups);
 
-        // Only store the header text if there was a parsing error or the it's less than 1MB on disk / 2MB in mem
-        if (!mFileHeader.getValidationErrors().isEmpty() || textHeader.length() < (1024 * 1024)) {
-            mFileHeader.setTextHeader(textHeader.toString());
-        }
-
         SAMUtils.processValidationErrors(mFileHeader.getValidationErrors(), -1, validationStringency);
         return mFileHeader;
     }
 
     private String advanceLine() {
         final int nextChar = mReader.peek();
-        if (nextChar != '@') {
-            return null;
-        }
-        mCurrentLine = mReader.readLine();
-        textHeader.append(mCurrentLine).append('\n');
-        return mCurrentLine;
+        this.mCurrentLine = (nextChar == '@') ? mReader.readLine() : null;
```

Comment:
It would be better if there was a named constant for the magic character/string `@`.